File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        res = float('inf')
        l = 0
        nums = sorted(nums)
        for r in range(k - 1, len(nums)):
            res = min(res, nums[r] - nums[l])
            l += 1
        return res

File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        if(k == 1)
        {
            return 0;
        }
        int res = INT_MAX;
        int l = 0;
        sort(nums.begin(), nums.end());
        for(int r = k - 1; r < nums.size(); r++)
        {
            res = min(res, nums[r] - nums[l]);
            l++;
        }
        return res;
    }
};